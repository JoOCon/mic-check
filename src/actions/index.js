export const addUser = (user) => ({
  type: 'ADD_USER',
  user
});

export const getLocalStorageUsers = (users) => ({
  type: 'GET_LOCAL_STORAGE_USERS',
  users
});

export const setActiveUser = (user) => ({
  type: 'SET_ACTIVE_USER',
  user
});

export const logoutActiveUser = () => ({
  type: 'LOGOUT_ACTIVE_USER'
});

export const getLocalStorageRentals = (rentals) => ({
  type: 'GET_LOCAL_STORAGE_RENTALS',
  rentals
});

export const addRental = (item) => ({
  type: 'ADD_RENTAL',
  item
});

export const removeRental = (item) => ({
  type: 'REMOVE_RENTAL',
  item
});

export const addQuote = (deliveryQuote) => ({
  type: 'ADD_QUOTE',
  deliveryQuote
});
