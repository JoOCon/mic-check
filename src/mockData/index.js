export const mockUser = {
  userName: 'Tom Hanks',
  password: 'password',
  email: 'THanks@hanks.com',
  location: '20 McAllister St, San Francisco, CA'
};

export const mockUsers = [{
  userName: 'Tom Hanks',
  password: 'password',
  email: 'THanks@hanks.com',
  location: '20 McAllister St, San Francisco, CA'
},
{
  userName: 'Jim Hanks',
  password: 'password',
  email: 'JHanks@hanks.com',
  location: '20 McAllister St, San Francisco, CA'
}];

export const mockItem = {
  name: 'Guitar',
  location: '101 Market St, San Francisco, CA',
  description: 'Its a guitar!',
  image: "https://guitar-image.png",
  id: 1
};

export const mockItem2 = {
  name: 'Amp',
  location: '101 Market St, San Francisco, CA',
  description: 'Its a amp!',
  image: "https://amp-image.png",
  id: 2
};

export const mockItems = [{
  name: 'Guitar',
  location: '101 Market St, San Francisco, CA',
  description: 'Its a guitar!',
  image: "https://guitar-image.png",
  id: 1
}, 
{
  name: 'Amp',
  location: '101 Market St, San Francisco, CA',
  description: 'Its a amp!',
  image: "https://amp-image.png",
  id: 2
}];

export const mockSenderLocation = '101 Market St, San Francisco, CA';
export const mockDeliveryLocation = '20 McAllister St, San Francisco, CA';

export const mockQuoteData = {
  kind: "delivery_quote",
  fee: 1140,
  created: "2018-09-13T00:40:02Z",
  pickup_duration: 5,
  expires: "2018-09-13T00:45:01Z",
  currency: "usd",
  currency_type: "USD",
  duration: 45,
  dropoff_eta: "2018-09-13T01:25:02Z",
  id: "dqt_Ltalsge5ktl1YV"
};

export const mockOrderInfo = {
  manifest: 'a box of shoes',
  'manifest_items[].name': 'Nikes',
  'manifest_items[].size': '2',
  pickup_address: '101 Market St, San Francisco, CA',
  pickup_name: 'Allen',
  pickup_phone_number: 3033334444,
  dropoff_address: '20 McAllister St, San Francisco, CA',
  dropoff_name: 'Tim',
  dropoff_phone_number: 2223334444,
  dropoff_notes: 'Tim@timmmy.com'
};