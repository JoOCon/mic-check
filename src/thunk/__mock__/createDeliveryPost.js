export const createDeliveryPost = jest.fn();