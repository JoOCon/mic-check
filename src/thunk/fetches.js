import { deliveryApiKey, customer_id  } from './apiKeys';

export const fetchQuote = async (fromAddress, toAddress) => {
  let addresses = {
    pickup_address: fromAddress,
    dropoff_address: toAddress
  };

  let deliveryBody = [];
  for (let property in addresses) {
    let encodedKey = encodeURIComponent(property);
    let encodedValue = encodeURIComponent(addresses[property]);
    deliveryBody.push(encodedKey + "=" + encodedValue);
  }
  deliveryBody = deliveryBody.join("&");

  const url = `https://api.postmates.com/v1/customers/${customer_id}/delivery_quotes`;
  const response = await fetch(url, {
    headers: {
      Authorization: "Basic " + btoa(deliveryApiKey + ":"),
      'Content-Type': 'application/x-www-form-urlencoded;charset=UTF-8'
    },
    method: 'POST',
    body: deliveryBody
  });
  const quote = await response.json();
  return quote;
};

export const fetchCompleteOrder = async (orderInfo) => {
  let orderObject = orderInfo;
  let deliveryBody = [];
  for (let property in orderObject) {
    let encodedKey = encodeURIComponent(property);
    let encodedValue = encodeURIComponent(orderObject[property]);
    deliveryBody.push(encodedKey + "=" + encodedValue);
  }
  deliveryBody = deliveryBody.join("&");

  const url = `https://api.postmates.com/v1/customers/${customer_id}/deliveries`;
  const response = await fetch(url, {
    headers: {
      'Authorization': "Basic " + btoa(deliveryApiKey + ":"),
      'Content-Type': 'application/x-www-form-urlencoded;charset=UTF-8'
    },
    method: 'POST',
    body: deliveryBody
  });
  const order = await response.json();
  return order;
};