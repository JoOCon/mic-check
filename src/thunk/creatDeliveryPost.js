import { deliveryApiKey, customer_id  } from './apiKeys';
import { addCompletedOrder } from '../actions';

export const creatDeliveryPost = (orderInfo) => {
  let orderObject = orderInfo;
  let deliveryBody = [];
  for (let property in orderObject) {
    let encodedKey = encodeURIComponent(property);
    let encodedValue = encodeURIComponent(orderObject[property]);
    deliveryBody.push(encodedKey + "=" + encodedValue);
  }
  deliveryBody = deliveryBody.join("&");

  return async (dispatch) => {
    const url = `https://api.postmates.com/v1/customers/${customer_id}/deliveries`;
    const response = await fetch(url, {
      headers: {
        'Authorization': "Basic " + btoa(deliveryApiKey + ":"),
        'Content-Type': 'application/x-www-form-urlencoded;charset=UTF-8'
      },
      method: 'POST',
      body: deliveryBody
    });
    const order = await response.json();
    dispatch(addCompletedOrder(order));
  };
};
