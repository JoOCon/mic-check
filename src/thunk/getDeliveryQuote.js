import { deliveryApiKey, customer_id  } from './apiKeys';
import { addQuote } from '../actions';
import PropTypes from 'prop-types';

export const getDeliveryQuote = (fromAddress, toAddress) => {
  return async (dispatch) => {
    const url = `https://api.postmates.com/v1/customers/${customer_id}/delivery_quotes`;
    const response = await fetch(url, {
      headers: {
        [deliveryApiKey]:'',
        'Content-Type': 'application/json'
      },
      method: 'Post',
      body: JSON.stringify({
        pickup_address: '20 McAllister St, San Francisco, CA',
        dropoff_address: '101 Market St, San Francisco, CA'
      })
    });
    const quote = await response.json();
    console.log(quote);
    dispatch(addQuote(quote));
  };
};

getDeliveryQuote.propTypes = {
  fromAddress: PropTypes.string,
  toAddress: PropTypes.string
};

// var form = new FormData();
// form.append("pickup_address", "20 McAllister St, San Francisco, CA");
// form.append("dropoff_address", "101 Market St, San Francisco, CA");

// var settings = {
//   "async": true,
//   "crossDomain": true,
//   "url": "https://api.postmates.com/v1/customers/cus_LsvzF5CZMxpsDk/delivery_quotes",
//   "method": "POST",
//   "headers": {
//     "Authorization": "Basic OWE4MTk4Y2QtZGVlZi00ZGIyLTk1NDAtM2ExMGJjYWE1Mzg0Og==",
//     "Cache-Control": "no-cache",
//     "Postman-Token": "5e01eac6-0ce1-499c-8a03-ed4d7ebf6222"
//   },
//   "processData": false,
//   "contentType": false,
//   "mimeType": "multipart/form-data",
//   "data": form
// };

// $.ajax(settings).done(function (response) {
//   console.log(response);
// });