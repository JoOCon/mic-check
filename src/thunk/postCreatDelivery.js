import { deliveryApiKey, customer_id  } from './apiKeys';
import { addQuote } from '../actions';

export const postCreatDelivery = (fromAddress, toAddress) => {
  let addresses = {
    pickup_address: fromAddress,
    dropoff_address: toAddress
  };

  var deliveryBody = [];
  for (var property in addresses) {
    var encodedKey = encodeURIComponent(property);
    var encodedValue = encodeURIComponent(addresses[property]);
    deliveryBody.push(encodedKey + "=" + encodedValue);
  }
  deliveryBody = deliveryBody.join("&");

  return async (dispatch) => {
    const url = `https://api.postmates.com/v1/customers/${customer_id}/delivery_quotes`;
    const response = await fetch(url, {
      headers: {
        'Authorization': "Basic " + btoa(deliveryApiKey + ":" + ""),
        'Content-Type': 'application/x-www-form-urlencoded;charset=UTF-8'
      },
      method: 'POST',
      body: deliveryBody
    });
    const quote = await response.json();
    dispatch(addQuote(quote));
  };
};
