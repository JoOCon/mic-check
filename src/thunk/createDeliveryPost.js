import { addCompletedOrder } from '../actions';
import { fetchCompleteOrder } from './fetches';

export const createDeliveryPost = (orderInfo) => {
  return async dispatch => {
    const order = await fetchCompleteOrder(orderInfo);
    dispatch(addCompletedOrder(order));
  };
};
