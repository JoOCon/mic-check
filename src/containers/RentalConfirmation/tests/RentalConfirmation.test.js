import React from 'react';
import { shallow } from 'enzyme';
import { RentalConfirmation, mapStateToProps, mapDispatchToProps } from '../';
import { createDeliveryPost } from '../../../thunk/createDeliveryPost';
import * as mockData from '../../../mockData';
import * as actions from '../../../actions';

jest.mock('../../../thunk/createDeliveryPost');

describe('RentalConfirmation tests', () => {
  let wrapper;
  let mockRunFunction;
  const { mockItem, mockUser, mockOrderInfo, mockQuoteData } = mockData;
  beforeEach(() => {
    mockRunFunction = jest.fn();
    wrapper = shallow( 
      <RentalConfirmation 
        quote={mockQuoteData}
        activeUser={mockUser}
        selectedItem={mockItem}
        createDeliveryPost={mockRunFunction}
        removeRental={mockRunFunction}
      /> 
    );
  });

  it('renders the RentalConfirmation with the correct props', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have initial state', () => {
    const mockInitalState = {
      redirect: false
    };
    expect(wrapper.state()).toEqual(mockInitalState);
  });

  describe('MapStateToProps', () => {
    it('should have a activeUser data object in props', () => {
      const mockState = {activeUser: mockUser};
      const expected = {activeUser: mockUser};
      const props = mapStateToProps(mockState);
      expect(props).toEqual(expected);
    });

    it('should have a quote data object in props', () => {
      const mockState = {quote: mockQuoteData};
      const expected = {quote: mockQuoteData};
      const props = mapStateToProps(mockState);
      expect(props).toEqual(expected);
    });

    it('should have a selectedItem data object in props', () => {
      const mockState = {selectedItem: mockItem};
      const expected = {selectedItem: mockItem};
      const props = mapStateToProps(mockState);
      expect(props).toEqual(expected);
    });
  });

  describe('MapDispatchToProps', () => {
    let mockDispatch;
    beforeEach(() => {
      mockDispatch = jest.fn();
    });

    it('should dispatch createDeliveryPost when called', () => {
      const mockAction = createDeliveryPost(mockOrderInfo);
      const props = mapDispatchToProps(mockDispatch);
      props.createDeliveryPost(mockOrderInfo);
      expect(mockDispatch).toBeCalledWith(mockAction);
    });

    it('should dispatch removeRental when called', () => {
      const { removeRental } = actions;
      const mockAction = removeRental(mockItem);
      const props = mapDispatchToProps(mockDispatch);
      props.removeRental(mockItem);
      expect(mockDispatch).toBeCalledWith(mockAction);
    });
  });
});
