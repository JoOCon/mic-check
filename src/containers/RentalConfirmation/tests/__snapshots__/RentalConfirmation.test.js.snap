// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RentalConfirmation tests renders the RentalConfirmation with the correct props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RentalConfirmation
    activeUser={
        Object {
            "email": "THanks@hanks.com",
            "location": "20 McAllister St, San Francisco, CA",
            "password": "password",
            "userName": "Tom Hanks",
          }
    }
    createDeliveryPost={[Function]}
    quote={
        Object {
            "created": "2018-09-13T00:40:02Z",
            "currency": "usd",
            "currency_type": "USD",
            "dropoff_eta": "2018-09-13T01:25:02Z",
            "duration": 45,
            "expires": "2018-09-13T00:45:01Z",
            "fee": 1140,
            "id": "dqt_Ltalsge5ktl1YV",
            "kind": "delivery_quote",
            "pickup_duration": 5,
          }
    }
    removeRental={[Function]}
    selectedItem={
        Object {
            "description": "Its a guitar!",
            "id": 1,
            "image": "https://guitar-image.png",
            "location": "101 Market St, San Francisco, CA",
            "name": "Guitar",
            "rentalPrice": 50,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          <h1>
                    Guitar
          </h1>
          <img
                    alt="rental item"
                    src="https://guitar-image.png"
          />
          <h1>
                    Item Info: 
                    Its a guitar!
          </h1>
          <h1>
                    Estimated Arrival: 
                    2018-09-13T01:25:02Z
          </h1>
          <h1>
                    Delivery Duration: 
                    45
                     minutes
          </h1>
          <h1>
                    Rental Cost: $
                    61.4
          </h1>
          <h1>
                    Offer Expires At: 
                    2018-09-13T00:45:01Z
          </h1>
          <button
                    onClick={[Function]}
          >
                    Place Order
          </button>
</div>,
        false,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              Guitar
</h1>,
            <img
              alt="rental item"
              src="https://guitar-image.png"
/>,
            <h1>
              Item Info: 
              Its a guitar!
</h1>,
            <h1>
              Estimated Arrival: 
              2018-09-13T01:25:02Z
</h1>,
            <h1>
              Delivery Duration: 
              45
               minutes
</h1>,
            <h1>
              Rental Cost: $
              61.4
</h1>,
            <h1>
              Offer Expires At: 
              2018-09-13T00:45:01Z
</h1>,
            <button
              onClick={[Function]}
>
              Place Order
</button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Guitar",
            },
            "ref": null,
            "rendered": "Guitar",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "rental item",
              "src": "https://guitar-image.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Item Info: ",
                "Its a guitar!",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Item Info: ",
              "Its a guitar!",
            ],
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Estimated Arrival: ",
                "2018-09-13T01:25:02Z",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Estimated Arrival: ",
              "2018-09-13T01:25:02Z",
            ],
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Delivery Duration: ",
                45,
                " minutes",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Delivery Duration: ",
              45,
              " minutes",
            ],
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Rental Cost: $",
                61.4,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Rental Cost: $",
              61.4,
            ],
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Offer Expires At: ",
                "2018-09-13T00:45:01Z",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Offer Expires At: ",
              "2018-09-13T00:45:01Z",
            ],
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Place Order",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Place Order",
            "type": "button",
          },
        ],
        "type": "div",
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <h1>
                        Guitar
            </h1>
            <img
                        alt="rental item"
                        src="https://guitar-image.png"
            />
            <h1>
                        Item Info: 
                        Its a guitar!
            </h1>
            <h1>
                        Estimated Arrival: 
                        2018-09-13T01:25:02Z
            </h1>
            <h1>
                        Delivery Duration: 
                        45
                         minutes
            </h1>
            <h1>
                        Rental Cost: $
                        61.4
            </h1>
            <h1>
                        Offer Expires At: 
                        2018-09-13T00:45:01Z
            </h1>
            <button
                        onClick={[Function]}
            >
                        Place Order
            </button>
</div>,
          false,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                Guitar
</h1>,
              <img
                alt="rental item"
                src="https://guitar-image.png"
/>,
              <h1>
                Item Info: 
                Its a guitar!
</h1>,
              <h1>
                Estimated Arrival: 
                2018-09-13T01:25:02Z
</h1>,
              <h1>
                Delivery Duration: 
                45
                 minutes
</h1>,
              <h1>
                Rental Cost: $
                61.4
</h1>,
              <h1>
                Offer Expires At: 
                2018-09-13T00:45:01Z
</h1>,
              <button
                onClick={[Function]}
>
                Place Order
</button>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Guitar",
              },
              "ref": null,
              "rendered": "Guitar",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "rental item",
                "src": "https://guitar-image.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Item Info: ",
                  "Its a guitar!",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Item Info: ",
                "Its a guitar!",
              ],
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Estimated Arrival: ",
                  "2018-09-13T01:25:02Z",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Estimated Arrival: ",
                "2018-09-13T01:25:02Z",
              ],
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Delivery Duration: ",
                  45,
                  " minutes",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Delivery Duration: ",
                45,
                " minutes",
              ],
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Rental Cost: $",
                  61.4,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Rental Cost: $",
                61.4,
              ],
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Offer Expires At: ",
                  "2018-09-13T00:45:01Z",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Offer Expires At: ",
                "2018-09-13T00:45:01Z",
              ],
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Place Order",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Place Order",
              "type": "button",
            },
          ],
          "type": "div",
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "disableLifecycleMethods": true,
    "hydrateIn": undefined,
  },
}
`;
